compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/KnownTargets.cpp

```diff
@@ -755,15 +755,25 @@ TargetAttr getHIPTargetDetails(StringRef target, StringRef features,
   return nullptr;
 }
 
-Attribute getHIPTargetEncodingLayoutAttr(TargetAttr target) {
-  // This is only enabled for CDNA2 and CDNA3 for the time being.
+Attribute getHIPTargetEncodingLayoutAttr(TargetAttr target,
+                                         StringRef resolver) {
+  if (resolver == kDataTilingEncodingLayoutResolverName) {
+    // Return a GPUEncodingLayoutAttr with an empty configuration. The addtional
+    // attributes will be attached by the `cloneWithSimplifiedConfig` interface
+    // method when the resolver needs to be configured.
+    return IREE::GPU::GPUEncodingLayoutAttr::get(target.getContext(), {});
+  }
+
+  // GPUPadLayoutAttr is only enabled for CDNA2 and CDNA3 for the time being.
   // TODO(kuhar): Enable for other HIP targets.
   if (!llvm::is_contained({"gfx90a", "gfx942"}, target.getArch())) {
     return nullptr;
   }
```

Comment:
IMO this condition should be sunk into the next if for it to make logical sense -- it's specific to the pad encoding resolver